### PR TITLE
Fix the documentation and remove last reference to mxnet

### PR DESCRIPTION
**Pull request by aarmey**
- The documentation is now built on each push, but only deployed on main.
- matplotlib was failing due to incompatibility with numpy 2.0
- Removed code that is already just installed from a package

### CODE REVIEW

**Comment by aarmey** [comment]
This should fix #547.

File: doc/_static/tensorly-pyramid.png
Comment (JeanKossaifi) [inline doc/_static/tensorly-pyramid.png:1]: Not sure what happened here? Was the file changed at all?

File: .github/workflows/build_doc.yml [hunk 1]
```diff
@@ -37,7 +34,7 @@ jobs:
           make html
           cd ..
       - name: Push doc to website
-        if: success()
+        if: success() && github.ref == 'refs/heads/main'
```
Comment (JeanKossaifi) [inline .github/workflows/build_doc.yml:37]: This is a great change, thank you!

**Review by JeanKossaifi — COMMENTED** [review]
This all looks great to me, thank you @aarmey !

File: doc/_static/tensorly-pyramid.png
Comment (aarmey) [reply doc/_static/tensorly-pyramid.png:1]: Yes, I removed mxnet.

**Pull request merged**